### PR TITLE
Fix uses of time.Truncate

### LINKED ISSUE #848: Audit all uses of `time.Truncate` in this codebase
### TL;DR
https://github.com/google/exposure-notifications-verification-server/issues/819

The above issue was found with stats, and Truncate is called throughout the codebase. Audit uses Truncate usage, and make sure we're doing the right thing.

### PR DESCRIPTION
**Pull request by jeremyfaller**
**** NB *****
This PR will be marked HOLD until all reviews requested on this PR give
it an LGTM. If you want to HOLD this CL, please add yourself as
reviewer, and mark yourself as NON-LGTM.
**** /NB *****

PTAL. These are all references to Truncate that:

`grep -R Truncate * | grep 24`

returned, and I think all changes are correct. Please verify.

Fixes #848

```release-note
Fixed statistics reporting dates.
Previously, it was possible for many statistics to report themselves on the wrong day. This has been fixed.
```

### CODE REVIEW

**Comment by jeremyfaller** [comment]
/hold

PTAL despite the hold.

File: pkg/database/vercode_test.go [hunk 1]
```diff
@@ -364,7 +364,7 @@ func TestStatDatesOnCreate(t *testing.T) {
 		{
 			var stats []*AuthorizedAppStats
 			if err := db.db.
-				Model(&UserStats{}).
+				Model(&AuthorizedAppStats{}).
```
Comment (sethvargo) [inline pkg/database/vercode_test.go:367]: This feels like it was a different bug?

**Review by sethvargo — APPROVED** [review]
/lgtm

File: pkg/database/authorized_app.go [hunk 2]
```diff
@@ -205,8 +206,8 @@ func (db *Database) FindAuthorizedAppByAPIKey(apiKey string) (*AuthorizedApp, er
 func (a *AuthorizedApp) Stats(db *Database, start, stop time.Time) ([]*AuthorizedAppStats, error) {
 	var stats []*AuthorizedAppStats
 
-	start = start.Truncate(24 * time.Hour)
-	stop = stop.Truncate(24 * time.Hour)
+	start = timeutils.Midnight(start)
```
Comment (mikehelmick) [inline pkg/database/authorized_app.go:209]: Should these be UTC?

File: pkg/database/realm.go [hunk 3]
```diff
@@ -1183,8 +1184,8 @@ type RealmUserStats struct {
 
 // CodesPerUser returns a set of UserStats for a given date range.
 func (r *Realm) CodesPerUser(db *Database, start, stop time.Time) ([]*RealmUserStats, error) {
-	start = start.Truncate(time.Hour * 24)
-	stop = stop.Truncate(time.Hour * 24)
+	start = timeutils.Midnight(start)
```
Comment (mikehelmick) [inline pkg/database/realm.go:1187]: utc?

**Review by mikehelmick — COMMENTED** [review]
added comments where the omissions of UTC is non-obvious to me

File: pkg/database/realm.go (hunk 3 above)
Comment (jeremyfaller) [reply pkg/database/realm.go]: Frankly, I am not sure. I've changed this (and above) to UTC, but I rely on other people who understand what we want our system to do more.

**Review by mikehelmick — COMMENTED** [review]
/lgtm

**Pull request merged**